File: Exercices/TGES/csharp_with_dp/LoggerMailObserver.cs
using System;
using System.Net.Mail;

using TheGreatEmailServer;

namespace Main
{
    public class LoggerMailObserver : MailObserver
    {
        public void MailReceived(MailMessage mail)
        {
            Console.WriteLine($"Received mail: {mail.From} {mail.To}");
        }

        public void MailRejected(MailMessage mail)
        {
            Console.WriteLine($"Rejected mail: {mail.From} {mail.To}");
        }

        public void MailStored(MailMessage mail)
        {
            Console.WriteLine($"Stored mail: {mail.From} {mail.To}");
        }
    }
}

File: Exercices/TGES/ruby/logger_mail_observer.rb
class LoggerMailObserver
  def mail_received(mail)
    puts "Received mail: #{mail.from} #{mail.to}"
  end
  
  def mail_rejected(mail)
    puts "Rejected mail: #{mail.from} #{mail.to}"
  end
  
  def mail_stored(mail)
    puts "Stored mail: #{mail.from} #{mail.to}"
  end
end
